crates/core/src/table/parquet_config.rs

```diff
@@ -0,0 +1,166 @@
+// Utilities to represent TableParquetOptions as a set of key-value pairs
+// Only active when the "datafusion" feature is enabled
+
+use std::collections::HashMap;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use std::fmt;
+use std::fmt::Display;
+use datafusion::common::{DataFusionError, file_options::parquet_writer::ParquetWriterOptions};
+use datafusion::config::{ConfigEntry, ConfigField, TableParquetOptions, Visit};
+use parquet::file::properties::WriterProperties;
+
+// String representation of the Parquet configuration.
+// This hashmap holds key-value pairs for the settings in
+// datafusion_common::config::TableParquetOptions
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ParquetConfig {
```

Comment:
If we don't need this string-serializable, I wonder if we could just directly use `TableParquetOptions` from DataFusion? Although delta-rs maintainers might reasonably prefer to not need to depend on DataFusion for this. I think you can write to a table without the datafusion feature enabled for example.